docs/features/incremental-generators.md

```diff
@@ -0,0 +1,478 @@
+# Incremental Generators
+
+## Summary
```

Comment:
Potential additions or follow-ups:
- when should I use an incremental generator vs. a plain source generator? (I assume the answer is the incremental is recommended)
- how do I convert my existing source generator to an incremental generator?